plugins/gateway/apigee/telemetry-policy.js

```diff
@@ -0,0 +1,117 @@
+// REQUEST
+//------>Header<------//
+var requestHeaderList = [];
+
+var requestHeaders = context.getVariable("request.headers.names"),
+
+requestResults = {};
+    
+requestHeaders = requestHeaders + '';
+
+requestHeaders = requestHeaders.slice(1, -1).split(', ');
+
+requestHeaders.forEach(function(k){
+  var v = context.getVariable("request.header." + k );
+  requestResults[k] = v;
+  var header = { key : k, value: v};
+  requestHeaderList.push(header);
+  
+});
+//------>Body<------//
+var reqContent = context.getVariable('request.content')
+
+if(reqContent !== null){
+    reqContent = crypto.base64(crypto.asBytes(reqContent))
+}
+
+
+var requestCommon = { 
+                        truncatedBody : true, 
+                        body: reqContent, 
+                        headers: requestHeaderList, 
+                        version: "0.0.1"
+                    };
+                    
+var req = { 
+            common : requestCommon, 
+            host: context.getVariable("request.header.host"), 
+            method: context.getVariable('request.verb'), 
+            path: context.getVariable("request.uri")
+          };
+
+
+// RESPONSE
+//------>Header<------//
+var responseHeaderList = [];
+
+var responseHeaders = context.getVariable("response.headers.names"),
+
+responseResults = {};
+
+responseHeaders = responseHeaders + '';
+
+responseHeaders = responseHeaders.slice(1, -1).split(', ');
+
+responseHeaders.forEach(function(k){
+  var v = context.getVariable("response.header." + k );
+  responseResults[k] = v;
+  var header = { key : k, value: v};
+  responseHeaderList.push(header);
+});
+
+//------>Body<------//
+var resContent = context.getVariable('response.content')
+
+if(resContent !== null){
+    resContent = crypto.base64(crypto.asBytes(resContent))
```

Comment:
body len need to be limited.
example https://github.com/apiclarity/apiclarity/blob/master/plugins/gateway/kong/plugin.go#L70